api/applicationoffers/client.go

```diff
@@ -73,48 +73,65 @@ func (c *Client) ListOffers(filters ...crossmodel.ApplicationOfferFilter) ([]cro
 		paramsFilter.Filters = append(paramsFilter.Filters, filterTerm)
 	}
 
-	applicationOffers := params.ListApplicationOffersResults{}
-	err := c.facade.FacadeCall("ListApplicationOffers", paramsFilter, &applicationOffers)
+	offers := params.QueryApplicationOffersResults{}
+	err := c.facade.FacadeCall("ListApplicationOffers", paramsFilter, &offers)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	return convertListResultsToModel(applicationOffers.Results)
+	return convertOffersResultsToModel(offers.Results)
 }
 
-func convertListResultsToModel(items []params.ApplicationOfferDetails) ([]crossmodel.ApplicationOfferDetailsResult, error) {
-	result := make([]crossmodel.ApplicationOfferDetailsResult, len(items))
+func convertOffersResultsToModel(items []params.ApplicationOfferAdminDetails) ([]*crossmodel.ApplicationOfferDetails, error) {
+	result := make([]*crossmodel.ApplicationOfferDetails, len(items))
+	var err error
 	for i, one := range items {
-		eps := make([]charm.Relation, len(one.Endpoints))
-		for i, ep := range one.Endpoints {
-			eps[i] = charm.Relation{
-				Name:      ep.Name,
-				Role:      ep.Role,
-				Interface: ep.Interface,
-			}
+		if result[i], err = offerParamsToDetails(one); err != nil {
+			return nil, errors.Trace(err)
 		}
-		result[i].Result = &crossmodel.ApplicationOfferDetails{
-			ApplicationName: one.ApplicationName,
-			OfferName:       one.OfferName,
-			CharmURL:        one.CharmURL,
-			OfferURL:        one.OfferURL,
-			Endpoints:       eps,
+	}
+	return result, nil
+}
+
+func offerParamsToDetails(offer params.ApplicationOfferAdminDetails) (*crossmodel.ApplicationOfferDetails, error) {
+	eps := make([]charm.Relation, len(offer.Endpoints))
+	for i, ep := range offer.Endpoints {
+		eps[i] = charm.Relation{
+			Name:      ep.Name,
+			Role:      ep.Role,
+			Interface: ep.Interface,
 		}
-		for _, oc := range one.Connections {
-			modelTag, err := names.ParseModelTag(oc.SourceModelTag)
-			if err != nil {
-				return nil, errors.Trace(err)
-			}
-			result[i].Result.Connections = append(result[i].Result.Connections, crossmodel.OfferConnection{
-				SourceModelUUID: modelTag.Id(),
-				Username:        oc.Username,
-				Endpoint:        oc.Endpoint,
-				RelationId:      oc.RelationId,
-				Status:          relation.Status(oc.Status.Status),
-				Message:         oc.Status.Info,
-				Since:           oc.Status.Since,
-				IngressSubnets:  oc.IngressSubnets,
-			})
+	}
+	result := &crossmodel.ApplicationOfferDetails{
+		ApplicationName:        offer.ApplicationName,
+		ApplicationDescription: offer.ApplicationDescription,
+		OfferName:              offer.OfferName,
+		CharmURL:               offer.CharmURL,
+		OfferURL:               offer.OfferURL,
+		Endpoints:              eps,
+		Access:                 permission.Access(offer.Access),
```

Comment:
The users block is only filled in if the person making the query is an admin - they get to see all users who can access the offer. The access attribute here is the access granted to the user making the query so that user can see if they have "read" or "consume" access etc. It's the same as for when you do a models query. You get to see your own access, and if admin, the access for other users also.

Reply:
I was wrong here - models only includes the current user i the user list